src/EditorFeatures/Core/StackTraceExplorer/StackFrameParserHelpers.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.StackTraceExplorer
+{
+    internal class StackFrameParserHelpers
+    {
+        /// <summary>
+        /// Makes sure that the string at least somewhat resembles the correct form.
+        /// Does not check validity on class or method identifiers
+        /// Example line:
+        /// at ConsoleApp4.MyClass.ThrowAtOne(p1, p2,) 
+        ///   |-------------------||--------||-------| 
+        ///           Class          Method    Args   
+        /// </summary>
+        /// <remarks>
+        /// See https://docs.microsoft.com/en-us/dotnet/api/system.environment.stacktrace for more information
+        /// on expected stacktrace form
+        /// </remarks>
+
+        public static bool TryParseMethodSignature(string line, int skipCharacters, out TextSpan classSpan, out TextSpan methodSpan, out TextSpan argsSpan)
+        {
+            Contract.ThrowIfTrue(skipCharacters < 0);
+
+            classSpan = default;
+            methodSpan = default;
+            argsSpan = default;
+
+            if (skipCharacters > 0)
+            {
+                line = line[skipCharacters..];
+            }
+
+            var regex = new Regex(@"(?<class>([a-zA-Z0-9_<>]+\.)+)(?<method>[a-zA-Z0-9_<>]+)\((?<args>.*)\).*");
```

Comment:
Have you tested generic methods and classes? I hate regex, so could be missing something, but with no backtick in the capture, how do you tell the difference between `Func<T>` and `Func<T, U>`?

Reply:
ooh, I don't have a test for multiple generics and it's likely broken. Let me add that... Likely wrong here. 

I don't love regex either. Thinking of a better way but this seemed the simplest so far. As it gets more complex it might be better to hand write

Reply:
Don't forget generic methods too :)

Also lambdas and async methods, or even just top level programs, which have unspeakable names.

Reply:
Unspeakable names I added as a not in scope for this PR, but will need to be done for final. 

I'll add tests for multiple configurations of generics though :)

Reply:
Added tests and fixed generics parsing in the regex. I don't love it, but it _works_ for now. Hopefully I'll discover a better way to parse these.